python/tvm/relay/backend/contrib/cmsisnn/codegen.py

```diff
@@ -0,0 +1,34 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Codegen for CMSIS-NN"""
+import tvm
+
+
+@tvm.register_func("relay.ext.cmsisnn")
+def cmsisnn_compiler(relay_func):
```

Comment:
Is there any reason to keep this in Python? I think this can be registered in C++ and the entire set of Python files removed now?

Reply:
Yes, we should move everything to c++ now.

Reply:
I thought this will go away with target hooks, so kept it this way for now.

Reply:
As a BYOC (which is what CMSIS integration is today), we dont need to maintain this in python especially everything else in C++. Please refer to other codegens in src/relay/backend/contrib/.

IMO, moving this to C++ does not have a dependency on target hooks.

Reply:
Ok, makes sense.